draft-regext-rdap-extensions.md

```diff
@@ -54,144 +54,236 @@ Domain Name Registries (DNRs), Regional Internet Registries
+
+### Profile and Marker Extensions {#profiles_and_markers}
+
+While the RDAP extension mechanism was created to extend RDAP queries
+and/or responses, extensions can also be used to signal server policy
+(for example, specifying the conditions of use for existing response
+structures). Extensions that are primarily about signaling server
+policy are often called "profiles".
+
+Some extensions exist to denote the usage of values placed into an
+IANA registry, such as the IANA RDAP registries, or the usage of
+extensions for specifications used in RDAP responses, such as extended
+vCard/jCard properties.  Such extensions exist to "mark" these usages
+and are often called "marker" extensions.
+
+For example, an extension may be used to signal desired processing of
+a "rel" attribute in a "links" array, where the "rel" value is
+registered in the Link Relations Registry
+(<https://www.iana.org/assignments/link-relations/link-relations.xhtml>):
+
+    {
+      "rdapConformance": [
+        "rdap_level_0",
+        "lunarNIC"
+      ],
+      "objectClassName": "domain",
+      "ldhName": "example.com",
+      "links": [
+        {
+          "value": "https://example.com/domain/example.com",
+          "href": "https://example.com/sideways_href",
+          "rel": "sideways",
+          "type": "application/rdap+json"
+        }
+      ]
+    }
+
+When defining the usage of link relations, extensions should specify
+the media types expected to be used with those link relations.
+
+Regardless of the category of the extension, its usage may also
+leverage the appearance of its identifier in the "rdapConformance"
+array.  Clients may use the "/help" query as defined in [@!RFC9082] to
+discover the extensions in use by the server.
+
+### Multiple Identifiers in Single Extension
+
+Extension specifications have customarily defined only one extension
+identifier.  However, there is no explicit limit on the number of
+extension identifiers that may be defined in a single extension
+specification.  The main reason for defining multiple identifiers is
+to reserve multiple namespaces in URLs or responses: see e.g.
+[@?I-D.ietf-regext-rdap-rir-search].
+
+## Syntax
+
+In brief, RDAP extension identifiers start with an alphabetic
+character and may contain alphanumeric characters and "_" (underscore)
+characters. This formulation was explicitly chosen to allow
+compatibility with variable names in programming languages and
+transliteration with XML.
+
+RDAP extension identifiers have no explicit structure, and are opaque
+insofar as no inner-meaning can be "seen" in them.  This document
+restricts the syntax of RDAP extension identifiers from containing two
+consecutive "_" (underscore) characters, reserving their use for the
+future definition of structure (such as for a versioning scheme). That
+is, RDAP extensions MUST NOT define an identifier with two consecutive
+underscore characters ("__") unless explicitly adhering to an RFC
+describing such usage.
```

Comment:
Since there is the Versioning draft being worked on, should be scratch out this paragraph? Pre-emptively reserving something being discussed somewhere else seems problematic.

Reply:
I think removing this text would be reasonable, given that the versioning document itself does not plan to use the two-underscore approach and that the chance of anybody registering anything problematic in the interim is low, but I'm also not concerned about leaving it as-is.  Will see what Andy thinks.

Reply:
Let's remove it. I reviewed the latest version of the versioning draft and it would appear from a client perspective they are all opaque.